### PR TITLE
support loading pushover secrets from files

### PR DESCRIPTION
**Pull request by sr**
Add the user_key_file and token_file keys to the pushover config.

/cc https://github.com/prometheus/alertmanager/issues/2498

Signed-off-by: Simon Rozet <me@simonrozet.com>

### CODE REVIEW

**Comment by sr** [comment]
Huum the test failure looks like a flake?

```
--- FAIL: TestSilences_Maintenance_SupportsCustomCallback (0.00s)
    silence_test.go:231: 
        	Error Trace:	/root/project/silence/silence_test.go:231
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 1
        	Test:       	TestSilences_Maintenance_SupportsCustomCallback
```

The previous commit passed:

https://app.circleci.com/pipelines/github/prometheus/alertmanager/3232

File: docs/configuration.md [hunk 1]
```diff
@@ -749,13 +749,16 @@ Pushover notifications are sent via the [Pushover API](https://pushover.net/api)
 # Whether to notify about resolved alerts.
 [ send_resolved: <boolean> | default = true ]
 
-# The recipient user's user key.
+# The recipient user's key. Either `user_key` or `user_key_file` should be set.
```
Comment (simonpasquier) [inline docs/configuration.md:752]:
```suggestion
# The recipient user's key. 
# user_key and user_key_file are mutually exclusive.
```

File: docs/configuration.md [hunk 2]
```diff
@@ -749,13 +749,16 @@ Pushover notifications are sent via the [Pushover API](https://pushover.net/api)
...
 user_key: <secret>
+user_key_file: <filepath>
 
 # Your registered application's API token, see https://pushover.net/apps
 # You can also register a token by cloning this Prometheus app:
 # https://pushover.net/apps/clone/prometheus
+# Either `token` or `token_file` should be set.
```
Comment (simonpasquier) [inline docs/configuration.md:759]:
```suggestion
# token and token_file are mutually exclusive.
```

**Review by simonpasquier — COMMENTED** [review]
lgtm. Just a few nits on the documentation to have a consistent wording.

**Comment by simonpasquier** [comment]
> Huum the test failure looks like a flake?

yep, nothing to do with your change.

**Comment by sr** [comment]
Thanks for the review, I've committed your suggestions.

**Comment by sr** [comment]
Do I need to do something to move this forward? Rebase?

**Comment by simonpasquier** [comment]
sorry for the late reply!

> Do I need to do something to move this forward? Rebase?

I'm investigating why the tests are flaky which could be resolved by #3216 . I'll let you know if you need to rebase after the other PR gets merged :)

**Comment by sr** [comment]
Sweet, thanks for fixing those flakey tests @simonpasquier 🥂

**Pull request merged**